test/range_mapper_tests.cc

```diff
@@ -128,102 +128,85 @@ TEST_CASE("slice built-in range mapper behaves as expected", "[range-mapper]") {
 	{
-		range_mapper rm{acc::all{}, access_mode::read, range<1>{128}};
+		range_mapper rm{acc::all{}, range<1>{128}};
 		auto sr = rm_result_to_subrange(rm.map_1(chunk<1>{}));
 		REQUIRE(sr.offset == id<1>{0});
 		REQUIRE(sr.range == range<1>{128});
 	}
 	{
-		range_mapper rm{acc::all{}, access_mode::read, range<2>{128, 64}};
+		range_mapper rm{acc::all{}, range<2>{128, 64}};
 		auto sr = rm_result_to_subrange(rm.map_2(chunk<1>{}));
 		REQUIRE(sr.offset == id<2>{0, 0});
 		REQUIRE(sr.range == range<2>{128, 64});
 	}
 	{
-		range_mapper rm{acc::all{}, access_mode::read, range<3>{128, 64, 32}};
+		range_mapper rm{acc::all{}, range<3>{128, 64, 32}};
 		auto sr = rm_result_to_subrange(rm.map_3(chunk<1>{}));
 		REQUIRE(sr.offset == id<3>{0, 0, 0});
 		REQUIRE(sr.range == range<3>{128, 64, 32});
 	}
 }
 
 TEST_CASE("neighborhood built-in range mapper behaves as expected", "[range-mapper]") {
-	SECTION("with shape = bounding_box") {
-		{
-			range_mapper rm{acc::neighborhood({10}), access_mode::read, range<1>{128}};
-			const auto r = rm.map_1(chunk<1>{{15}, {10}, {128}});
-			CHECK(r == box<1>(5, 35));
-		}
-		{
-			range_mapper rm{acc::neighborhood({10, 10}), access_mode::read, range<2>{128, 128}};
-			const auto r = rm.map_2(chunk<2>{{5, 100}, {10, 20}, {128, 128}});
-			CHECK(r == box<2>({0, 90}, {25, 128}));
-		}
-		{
-			range_mapper rm{acc::neighborhood({3, 4, 5}), access_mode::read, range<3>{128, 128, 128}};
-			const auto r = rm.map_3(chunk<3>{{3, 4, 5}, {1, 1, 1}, {128, 128, 128}});
-			CHECK(r == box<3>({0, 0, 0}, {7, 9, 11}));
-		}
-	}
-
-	SECTION("with shape = along_axes") {
-		{
-			range_mapper rm{acc::neighborhood({10}, neighborhood_shape::along_axes), access_mode::read, range<1>{128}};
-			const auto r = rm.map_1(chunk<1>{{15}, {10}, {128}});
-			CHECK(r == box<1>({5}, {35}));
-		}
-		{
-			range_mapper rm{acc::neighborhood({10, 10}, neighborhood_shape::along_axes), access_mode::read, range<2>{128, 128}};
-			const auto r = rm.map_2(chunk<2>{{5, 100}, {10, 20}, {128, 128}});
-			CHECK(r == region<2>({box<2>({0, 100}, {25, 120}), box<2>({5, 90}, {15, 128})}));
-		}
-		{
-			range_mapper rm{acc::neighborhood({3, 4, 5}, neighborhood_shape::along_axes), access_mode::read, range<3>{128, 128, 128}};
-			const auto r = rm.map_3(chunk<3>{{3, 4, 5}, {1, 1, 1}, {128, 128, 128}});
-			CHECK(r == region<3>({box<3>({0, 4, 5}, {7, 5, 6}), box<3>({3, 0, 5}, {4, 9, 6}), box<3>({3, 4, 0}, {4, 5, 11})}));
-		}
+	{
+		range_mapper rm{acc::neighborhood<1>({10}), range<1>{128}};
+		auto sr = rm_result_to_subrange(rm.map_1(chunk<1>{{15}, {10}, {128}}));
+		REQUIRE(sr.offset == id<1>{5});
+		REQUIRE(sr.range == range<1>{30});
+	}
+	{
+		range_mapper rm{acc::neighborhood<2>({10, 10}), range<2>{128, 128}};
+		auto sr = rm_result_to_subrange(rm.map_2(chunk<2>{{5, 100}, {10, 20}, {128, 128}}));
+		REQUIRE(sr.offset == id<2>{0, 90});
+		REQUIRE(sr.range == range<2>{25, 38});
+	}
+	{
+		range_mapper rm{acc::neighborhood<3>({3, 4, 5}), range<3>{128, 128, 128}};
```

Comment:
You seem to have reverted the changes from #292 here?

Reply:
Oops, since the tests moved to a new file I missed that while copying over the changes.